### ISSUE TITLE
Move screen selection sounds to joist

### ISSUE THREAD

**Issue by jbphet** [phetsims/tambo#91]
Once the screen selection sounds have been finalized, they should be moved into joist.  For now, they are being played in the tambo demo only.

**Comment by samreid** [comment]
In the above commit, I moved the `ScreenSelectionSoundGenerator` instantiation to joist so it can be used in any sim with sound.  I did not yet move the sound files or `ScreenSelectionSoundGenerator` itself.  @jbphet can you please review?  It seems to me that `ScreenSelectionSoundGenerator` should move to joist as well, but I wasn't sure about the sounds (whether they would be reused elsewhere).

**Comment by samreid** [comment]
This is crashing single-screen sims with audio, I'll push a workaround momentarily.

**Comment by samreid** [comment]
I added a workaround, but I wasn't really sure what caused the problem in the first place.  @jbphet can you please investigate further?

**Comment by jbphet** [comment]
@samreid - I removed the workaround as a test, and it seemed to work fine (no crashing of single-screen sims with audio), so it might be that things have changed due to the way sounds work after the ES6 migration.  However, I'm inclined to leave the conditional in place, since the sound generator wouldn't be used in a single-screen sim, and not creating it will save load time.  Are you okay with leaving it as is?

Two other things worth noting:
1. Part of the comment in this area of the code, currently on line 804 of `Sim.js`, may no longer be true once https://github.com/phetsims/tambo/issues/100 is addressed.  Am I correct on this?  I so, let me know and I'll make a note in the tambo issue to clean up the comment.
1. I moved the sound generator and the associated files into joist, since these items are now specific to this repo.  Please have a look and make sure you're okay with what I've done.

**Comment by samreid** [comment]
> However, I'm inclined to leave the conditional in place, since the sound generator wouldn't be used in a single-screen sim, and not creating it will save load time. Are you okay with leaving it as is?

The move to joist looks like it was done nicely and is working well.  However, it seems a more appropriate way to set this up would be to add the `ScreenSelectionSoundGenerator` as part of the `HomeScreenView`.  That way it will only be created if there is a home screen, and it won't need to be synchronized with `workItems` etc.

The last part of the comment says:
> and needs a timeout afterwards to make sure that the sound urls can load (thus before the final workItem).

which is no longer necessary now that #100 is complete.

I tried for 20+ minutes to move `ScreenSelectionSoundGenerator` to `HomeScreenView` but I wasn't familiar enough with the implementation to proceed confidently.  Want to close this issue "about moving things to joist" and open a separate issue for moving `ScreenSelectionSoundGenerator` to `HomeScreenView`?

**Comment by jbphet** [comment]
Since I'm going to be making changes to the code that does the sound generation for screen selection, I thought it would be a good idea to publish a dev version of multi-screen sim that demonstrates the current behavior as a reference point.  Here it is: https://phet-dev.colorado.edu/html/tambo/1.0.0-dev.14/phet/tambo_en_phet.html.  Once the refactoring is done, the behavior should be the same as in this reference version.

**Comment by jbphet** [comment]
@samreid said:

>  [I]t seems a more appropriate way to set this up would be to add the `ScreenSelectionSoundGenerator` as part of the `HomeScreenView`...

Good suggestion.  I made this change and it enabled some substantial simplifications to the code.  @samreid - please review.  If you're cool with what we have at this point, feel free to close.

**Comment by samreid** [comment]
The change set looks really great, thanks!  Closing.
